src/shared_modules/router/tests/component/interface_c_test.cpp

```diff
@@ -77,3 +77,38 @@ TEST_F(RouterCInterfaceTest, TestProviderSendZero)
 
     // TO DO - Add C interface for subscribers.
 }
+
+TEST_F(RouterCInterfaceTest, TestProviderSendAndDestroy)
+{
+    auto handle = router_provider_create("test");
+
+    EXPECT_NE(handle, nullptr);
+
+    EXPECT_EQ(router_provider_send(handle, "test", 4), 0);
+
+    EXPECT_NO_THROW(router_provider_destroy(handle));
+
+    // TO DO - Add C interface for subscribers.
```

Comment:
Nitpick: most IDEs detect TODO comments like this
```suggestion
    // TODO - Add C interface for subscribers.
```

Reply:
done